lib/elasticband/filter/range.rb

```diff
@@ -0,0 +1,29 @@
+module Elasticband
+  class Filter
+    class Range < Filter
+      attr_accessor :field, :ranges
+
+      RANGES = %i(gt gteq lt lteq time_zone)
+
+      def initialize(field, ranges)
+        self.field = field.to_sym
+        self.ranges = permitted_ranges(ranges)
+      end
+
+      def to_h
+        { range: { field => parsed_ranges } }
+      end
+
+      def permitted_ranges(ranges)
+        ranges.keep_if { |key, _| RANGES.include?(key) }
+      end
+
+      def parsed_ranges
+        ranges[:gte] = ranges.delete(:gteq) if ranges.key?(:gteq)
```

Comment:
As you are deleting element from this Hash, and therefore modifying it, wouldn't be better to dup it? This can be used by the user later on ;)